fontbe/src/gvar.rs

```diff
@@ -0,0 +1,101 @@
+//! Generates a [gvar](https://learn.microsoft.com/en-us/typography/opentype/spec/gvar) table.
+
+use std::cmp::{min, max};
+
+use font_types::{F2Dot14, GlyphId};
+use fontdrasil::orchestration::Work;
+use fontir::coords::NormalizedLocation;
+use write_fonts::{
+    dump_table,
+    tables::{
+        gvar::{GlyphDeltas, GlyphVariations, Gvar},
+        variations::Tuple,
+    },
+};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Bytes, Context, GvarFragment},
+};
+
+struct GvarWork {}
+
+pub fn create_gvar_work() -> Box<BeWork> {
+    Box::new(GvarWork {})
+}
+
+fn to_tuple(loc: &NormalizedLocation) -> Tuple {
+    Tuple::new(
+        loc.iter()
+            .map(|(_, nc)| F2Dot14::from_f32(nc.to_f32()))
+            .collect(),
+    )
+}
+
+fn to_deltas(fragment: &GvarFragment) -> Vec<GlyphDeltas> {
```

Comment:
I believe the order in which you collect the vector of deltas is not deterministic, because VariationModel::deltas currently returns a `HashMap<NormalizedLocation, Vec<V>>` (whose order is not insertion-order, nor ordered, just random I believe); in fonttools VariationModel.getDeltas returns an ordered list of deltas, and the order matches the internal ordering of the master locations (as sorted by the variation model) and this order is deterministic. We want to keep that order, so maybe instead of returning a `HashMap<NormalizedLocation, Vec<V>>` from `deltas` method, you should return a vector or iterator of tuples `(NormalizedLocation, Vec<V>)`.
Even better, as I was saying in https://github.com/googlefonts/fontmake-rs/pull/236/files#r1159650526, the `deltas` method could return the regions instead of the locations for each deltas, i.e. `(VariationRegion, Vec<V>)`.

Reply:
probably the rendering won't change if TupleVariations for a given glyph are ordered differently, since in the end the final set of deltas is going to result from a summation of the respective (scaled) deltas; but keeping the same order as fonttools and not throwing away the deterministic ordering that we already have are good things by themselves.

Reply:
I think I had in mind the order didn't matter. I agree that's not going to be great wrt matching FontTools. Will change to something ordered.